include/eve/module/real/core/function/regular/simd/x86/convert_256.hpp

```diff
@@ -30,6 +30,14 @@ namespace eve::detail
     {
       return v0;
     }
+    else if constexpr( current_api >= avx2 && (N::value == 4)
+                       && std::is_integral_v<In> && (sizeof(In) == 8)
+                       && std::is_integral_v<Out> && (sizeof(Out) == 4) )
```

Comment:
Hmm, what about signed unsigned?

Should probably be OK, I dunno what we have in terms of tests here.

This covers I guess: https://github.com/jfalcou/eve/blob/d1885fd6e0849899ab070ab35bb313aafbf187a1/test/unit/module/real/core/convert/to_int32.cpp#L60-L71

I did check - the intrinsics, I think this shuffle is your best bet unfortunatly.

Reply:
IMO signed and unsigned are both covered by this case. Indeed, for me a downcast of s64 => s32 is the same as u64 => u32, thanks to the two complements representation.

Reply:
For instance, LLVM has a `zext` and `sext` operation, but only a `trunc` one: https://llvm.org/docs/LangRef.html#conversion-operations